core/src/main/kotlin/org/evomaster/core/EMConfig.kt

```diff
@@ -575,4 +609,112 @@ class EMConfig {
+
+    @Experimental
+    @Cfg("Specify whether to enable resource-based strategy to sample an individual during search. " +
+            "Note that resource-based sampling is only applicable for REST problem with MIO algorithm.")
+    var resourceSampleStrategy = ResourceSamplingStrategy.NONE
+
+    @Experimental
+    @Cfg("Specify whether to enable resource dependency heuristics, i.e, probOfEnablingResourceDependencyHeuristics > 0.0. " +
+            "Note that the option is available to be enabled only if resource-based smart sampling is enable. " +
+            "This option has an effect on sampling multiple resources and mutating a structure of an individual.")
+    @Min(0.0) @Max(1.0)
+    var probOfEnablingResourceDependencyHeuristics = 0.0
+
+    @Experimental
+    @Cfg("Specify whether to export derived dependencies among resources. " +
+            "if enabled, the dependencies are stored at the same location of generated test cases.")
+    var exportDependencies = false
+
+    @Experimental
+    @Cfg("Specify a file that saves derived dependencies")
+    var dependencyFile = "dependencies.csv"
+
+    @Experimental
+    @Cfg("Whether to involve database when applying resource-based methods, e.g., resource-based sampling, dependency analysis between resource and tables.")
+    var doesInvolveDatabase = false
+
+    @Experimental
+    @Cfg("Specify a minimal number of rows in a table that enables selection (i.e., SELECT sql) to prepare resources for REST Action. In other word, if the number is less than the specified, insertion is always applied.")
+    @Min(0.0)
+    var minRowOfTable = 10
+
+    @Experimental
+    @Cfg("Specify a probability that enables selection (i.e., SELECT sql) of data from database instead of insertion (i.e., INSERT sql) for preparing resources for REST actions")
+    @Min(0.0) @Max(1.0)
+    var probOfSelectFromDatabase = 0.1
+
+    @Experimental
+    @Cfg("Whether to apply text/name analysis with natural la parser to derive relationships between name entities, e.g., a resource identifier with a name of table")
+    var doesApplyNameMatching = false
+
+    @Experimental
+    @Cfg("Specify a probability to apply S1iR when sample control is Customized")
+    @Min(0.0)@Max(1.0)
+    var S1iR : Double = 0.25
+
+    @Experimental
+    @Cfg("Specify a probability to apply S1dR when sample control is Customized")
+    @Min(0.0)@Max(1.0)
+    var S1dR : Double = 0.25
+
+    @Experimental
+    @Cfg("Specify a probability to apply S2dR when sample control is Customized")
+    @Min(0.0)@Max(1.0)
+    var S2dR : Double = 0.25
+
+    @Experimental
+    @Cfg("Specify a probability to apply SMdR when sample control is Customized")
+    @Min(0.0)@Max(1.0)
+    var SMdR : Double = 0.25
+
+    @Experimental
+    @Cfg("Specify a probability to enable archive-based mutation")
+    @Min(0.0) @Max(1.0)
+    var probOfArchiveMutation = 0.0
+
+    @Experimental
+    @Cfg("Specify a percentage to prepare candidates of genes to mutate")
```

Comment:
I do not understand this one. Can you please add a bit more description?

Reply:
fixed, thanks